client/my-sites/domains/domain-management/settings/set-as-primary/types.ts

```diff
@@ -0,0 +1,33 @@
+import {
+	showUpdatePrimaryDomainSuccessNotice,
+	showUpdatePrimaryDomainErrorNotice,
+} from 'calypso/state/domains/management/actions';
+import { setPrimaryDomain } from 'calypso/state/sites/domains/actions';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { SiteData } from 'calypso/state/ui/selectors/site-data';
+
+export type SetAsPrimaryPassedProps = {
+	domain: ResponseDomain;
+	selectedSite: SiteData;
+};
+
+export type SetAsPrimaryStateProps = {
+	canSetPrimaryDomain: boolean;
+	hasNonPrimaryDomainsFlag: boolean;
+	hasDomainOnlySite: boolean;
+	isOnFreePlan: boolean;
+	isManagingAllSites: boolean;
+};
+
+export type ChangePrimaryFunctionSignature = ( domain: ResponseDomain, mode: string ) => void;
```

Comment:
Nitpick: naming this type as `FunctionSignature` seems a little redundant 🙂 I think this could be renamed to `ChangePrimaryFunction`